smithy-codegen-core/src/main/java/software/amazon/smithy/codegen/core/Definitions.java

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.smithy.codegen.core;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.InputStream;
+import java.net.URI;
+import java.util.HashMap;
+import java.util.Map;
+
+import software.amazon.smithy.model.node.Node;
+import software.amazon.smithy.model.node.NodeMapper;
+import software.amazon.smithy.model.node.ObjectNode;
+import software.amazon.smithy.model.node.ToNode;
+import software.amazon.smithy.utils.MapUtils;
+import software.amazon.smithy.utils.SmithyBuilder;
+import software.amazon.smithy.utils.ToSmithyBuilder;
+
+/**
+ * Class that defines the acceptable values that can be used in ShapeLink objects.
+ * <p>
+ * The types {@link Map} defines the set of types that can be used in a {@link ShapeLink} object.
+ * Each key is the name of the type, and each value is a description of the type.
+ * For programming languages, these types represent language-specific components.
+ * For example, in Java, types should map to the possible values of {@link java.lang.annotation.ElementType}.
+ * </p>
+ * <p>
+ * The tags {@link Map} defines the set of tags that can be used in a {@link ShapeLink} object.
+ * Each key is the name of the tag, and each value is the description of the tag.
+ * Tags are used to provide semantics to links. Tools that evaluate trace models
+ * use these tags to inform their analysis. For example, a tag for an AWS SDK code
+ * generator could be "requestBuilder" to indicate that a class is used as a builder for a request.
+ * </p>
+ */
+public final class Definitions implements ToNode, ToSmithyBuilder<Definitions> {
```

Comment:
`Definitions` seems slightly too generic of a name for this class, unless you think these will be useful beyond ShapeLink objects.